### PR TITLE
Custom fields addition for plans, addons, customers, subscriptions

### PR DESCRIPTION
**Pull request by cb-senthilvel**

### CODE REVIEW

File: tap_chargebee/streams/base.py [hunk 1]
```diff
@@ -71,11 +71,42 @@ def generate_catalog(self):
...
+            content = record['content']
+            words = record['event_type'].split("_")
+            counter = 1;
+            content_obj = ""
+            for word in words:
+                if counter != len(words):
+                    content_obj = str(content_obj) + word 
+                    if(counter + 1 != len(words)):
+                        content_obj = content_obj + "_"
+                counter = counter + 1
```
Comment (dmosorast) [inline tap_chargebee/streams/base.py]: Can this be made more clear with a unit test? This seems like it's either doing nothing or rebuilding the original `event_type` string, with the last 1 or 2 sections removed. A string split, slice, and join would be more clear in the latter case.

File: tap_chargebee/streams/base.py [hunk 2]
```diff
@@ -128,7 +159,29 @@ def sync_data(self):
...
+                        Util.coupons.append(event['content']['coupon'])
+            if self.ENTITY == 'plan':
+                for plan in Util.plans:
+                    to_write.append(plan)
+            if self.ENTITY == 'addon':
+                for addon in Util.addons:
+                    to_write.append(addon)
+            if self.ENTITY == 'coupon':
+                for coupon in Util.coupons:
+                    to_write.append(coupon)
```
Comment (dmosorast) [inline tap_chargebee/streams/base.py]: Was this included intentionally? Seems to not be related to the rest of the PR.

Comment (cb-senthilvel) [reply tap_chargebee/streams/base.py]: It seems while merging it is added. Now removed.

File: tap_chargebee/streams/base.py (hunk 1 above)
Comment (cb-senthilvel) [reply tap_chargebee/streams/base.py]: From events_type( a field in Event stream's record) we are fetching the main object for which, this event triggered (sample value "credit_note_updated" ==> "credit_note").  Full resource of that object will be present in the event object. The resource may have the custom fields and we are constructing all the custom fields as a single JSON field.

Changed the code as you mentioned.

**Comment by dmosorast** [comment]
Looks good, that makes more sense. Thanks! I'm good to merge and release this change.

**Pull request merged**